src/supertux/flip_level_transformer.cpp

```diff
@@ -30,9 +31,20 @@ void
 FlipLevelTransformer::transform_sector(Sector& sector)
 {
   float height = sector.get_height();
+  std::set<Path*> paths;
 
   for (auto& object : sector.get_objects()) {
     object->on_flip(height);
+    PathObject* po = dynamic_cast<PathObject*>(object.get());
+    if (po && po->get_path()) {
+      if (paths.find(po->get_path()) == paths.end()) {
+        paths.insert(po->get_path());
+      }
+    }
+  }
+
+  for (Path* path : paths) {
+    path->on_flip(height);
   }
 
```

Comment:
I'm sorry for only mentioning this now, I just thought about it:

Would it not be easier to put this code in the `PathGameObject` class? That class holds directly the interesting paths, and putting the relevant code there removes the needs for:
- `dynamic_cast`'ing and treating PathObjects specially;
- check for duplicate paths;
- adding code in every class that makes use of a path.